lib/generate-form-element.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+'use strict'
+
+const uuid = require('uuid/v4')
+
+const { validateWithElementSchema } = require('../lib/forms-validation.js')
+
+async function generateFormElement (
+  formElementGenerationData /* : mixed */
+) /* : Promise<FormElement> */ {
```

Comment:
Can we change this so that it does not return a promise, most developers will not want to deal promises when mapping there data to form elements.

Joi's [validate](https://github.com/hapijs/joi/blob/master/API.md#validatevalue-schema-options-callback) function does not required a callback. Could we make this change to the `validateForm()` function as well.